src/libNetwork/P2PComm.cpp

```diff
@@ -583,6 +606,81 @@ void P2PComm::EventCallback(struct bufferevent* bev, short events,
         m_dispatcher(raw_message);
     }
+    else if (startByte == START_BYTE_GOSSIP)
+    {
+        // Check for length consistency
+
+        if (messageLength != message.size() - HDR_LEN)
+        {
+            LOG_GENERAL(WARNING, "Incorrect message length.");
+            return;
+        }
+
+        if (messageLength < GOSSIP_MSGTYPE_LEN + GOSSIP_ROUND_LEN
+                + GOSSIP_SNDR_LISTNR_PORT_LEN)
+        {
+            LOG_GENERAL(
+                WARNING,
+                "Gossip Msg Type and/or Gossip Round and/or SNDR LISTNR "
+                "Port is missing (messageLength = "
+                    << messageLength << ")");
+            return;
+        }
+
+        std::vector<unsigned char> gossipMsgTyp(message.begin() + HDR_LEN,
+                                                message.begin() + HDR_LEN
+                                                    + GOSSIP_MSGTYPE_LEN);
+
+        std::vector<unsigned char> tmp(
+            message.begin() + HDR_LEN + GOSSIP_MSGTYPE_LEN,
+            message.begin() + HDR_LEN + GOSSIP_MSGTYPE_LEN + GOSSIP_ROUND_LEN);
+
+        const uint32_t gossipMsgRound
+            = (tmp[0] << 24) + (tmp[1] << 16) + (tmp[2] << 8) + tmp[3];
+
+        tmp.clear();
+        tmp.insert(tmp.end(),
+                   message.begin() + HDR_LEN + GOSSIP_MSGTYPE_LEN
+                       + GOSSIP_ROUND_LEN,
+                   message.begin() + HDR_LEN + GOSSIP_MSGTYPE_LEN
+                       + GOSSIP_ROUND_LEN + GOSSIP_SNDR_LISTNR_PORT_LEN);
+
+        const uint32_t gossipSenderPort
+            = (tmp[0] << 24) + (tmp[1] << 16) + (tmp[2] << 8) + tmp[3];
+        from.m_listenPortHost = gossipSenderPort;
+
+        RumorManager::RawBytes rumor_message(
+            message.begin() + HDR_LEN + GOSSIP_MSGTYPE_LEN + GOSSIP_ROUND_LEN
+                + GOSSIP_SNDR_LISTNR_PORT_LEN,
+            message.end());
+
+        P2PComm& p2p = P2PComm::GetInstance();
+        if (gossipMsgTyp[0] == (uint8_t)RRS::Message::Type::FORWARD)
+        {
+            LOG_GENERAL(
+                INFO, "Received Gossip of type - FORWARD from Peer :" << from);
+
+            if (p2p.SpreadRumor(rumor_message))
+            {
+                std::pair<vector<unsigned char>, Peer>* raw_message
+                    = new pair<vector<unsigned char>, Peer>(rumor_message,
+                                                            from);
+
+                // Queue the message
+                m_dispatcher(raw_message);
+            }
+        }
+        else if (p2p.m_rumorManager.RumorReceived((unsigned int)gossipMsgTyp[0],
+                                                  gossipMsgRound, rumor_message,
+                                                  from))
+        {
+            std::pair<vector<unsigned char>, Peer>* raw_message
+                = new pair<vector<unsigned char>, Peer>(rumor_message, from);
+
+            // Queue the message
+            m_dispatcher(raw_message);
+        }
```

Comment:
Perhaps we can add an else case here, and log an error msg.

Reply:
Its not error in general but just that we don't need to dispatch it. Reason for not dispatching is already logged in RumorReceived() for ex: when gossip is old one.